src/iuse.cpp

```diff
@@ -8493,13 +8493,13 @@ int iuse::saw_barrel( player *p, item *, bool, const tripoint& )
         return 0;
     }
 
-    auto filter = [&]( const item& e ) {
+    auto filter = []( const item& e ) {
         if( !e.is_gun() || e.type->gun->barrel_length <= 0 ) {
             return false;
         }
         // cannot saw down barrel of gun that already has a barrel mod
-        return std::none_of( e.contents.begin(), e.contents.end(), [&]( const item& mod ) {
-            return mod.type->gunmod->location == "barrel";
+        return std::none_of( e.contents.begin(), e.contents.end(), []( const item& mod ) {
```

Comment:
I see no advantage. It requires constructing yet another vector and doubles iteration costs, but without any sensible benefit: it doesn't reduce amount of code and actually increases it. Clarity is arguable. The only strong reason for doing so is encapsulating `contents` (or storing gunmods somewhere else in the future), but major goal of the PR was fixing the crash, not refactoring your code.